File: app/src/main/java/com/arcao/wherigoservice/api/parser/WherigoJsonResultParser.java
package com.arcao.wherigoservice.api.parser;


import com.arcao.geocaching.api.parser.JsonReader;

import java.io.IOException;

public class WherigoJsonResultParser {
    public static Result parse(JsonReader r) throws IOException {
        Result status = new Result();
        r.beginObject();
        while (r.hasNext()) {
            String name = r.nextName();
            switch (name) {
                case "Code":
                    status.statusCode = r.nextInt();
                    break;
                case "Text":
                    status.statusMessage = r.nextString();
                    break;
                default:
                    r.skipValue();
                    break;
            }
        }
        r.endObject();
        return status;
    }

    public static class Result {
        int statusCode;
        String statusMessage;

        public int getStatusCode() {
            return statusCode;
        }

        public String getStatusMessage() {
            return statusMessage;
        }
    }
}

File: app/src/main/java/com/arcao/wherigoservice/api/parser/WherigoJsonResultParser.kt
package com.arcao.wherigoservice.api.parser

import com.google.gson.stream.JsonReader

import java.io.IOException

object WherigoJsonResultParser {
    @Throws(IOException::class)
    fun parse(r: JsonReader): Result {
        val status = Result()
        r.beginObject()
        while (r.hasNext()) {
            when (r.nextName()) {
                "Code" -> status.statusCode = r.nextInt()
                "Text" -> status.statusMessage = r.nextString()
                else -> r.skipValue()
            }
        }
        r.endObject()
        return status
    }

    class Result {
        var statusCode: Int = 0
            internal set
        var statusMessage: String? = null
            internal set
    }
}
